sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -185,6 +209,12 @@ private void stopTracing(final @NonNull Activity activity, final boolean shouldF
 
   private void finishTransaction(final @Nullable ITransaction transaction) {
     if (transaction != null) {
+      // if io.sentry.traces.activity.auto-finish.enable is disabled, transaction may be already
+      // finished when this method is called.
+      if (transaction.isFinished()) {
```

Comment:
`l`: It's good that we check that here, but I hope it isn't a problem when you call finish on a finished transaction. Maybe we don't need that check here.

Reply:
this is needed since `stopPreviousTransactions` calls this method, the comment explains why its needed